sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -754,6 +754,8 @@ case class AlterTableSetLocationCommand(
         // No partition spec is specified, so we set the location for the table itself
         catalog.alterTable(table.withNewStorage(locationUri = Some(location)))
     }
+
+    catalog.refreshTable(table.identifier)
```

Comment:
This is not a right fix. Please see the discussion in a separate PR: https://github.com/apache/spark/pull/16514#issuecomment-271232337

Please focus on the scope of this PR. Thanks!